### PR TITLE
[Load test iot simulator]Use variable group, align resource names with IaC, add readme

### PR DESCRIPTION
**Pull request by shervyna**
## Purpose
- use variable group for keyvault secrets, to align with the IaC pipeline, and being able to skip the step to add access policy in keyvault manually.
- update the variable names( eg. resource group name) to align with the resources created by iac.
- added comments in the yaml to explain more in detail
- added readme with setup details
- changed the default load in powershell script to larger load

## Does this introduce a breaking change?

- [X] Yes
- [ ] No

## Author pre-publish checklist:

- [ ] Added test to prove my fix is effective or new feature works
- [X] No PII in logs
- [X] Made corresponding changes to the documentation

### CODE REVIEW

**Comment by jmostella** [comment]
Can we have more of a description in the title?
Also can you detail which issues are fixed in the PR comments?

**Review by jmostella — CHANGES_REQUESTED** [review]

**Comment by shervyna** [comment]
> Can we have more of a description in the title?
> Also can you detail which issues are fixed in the PR comments?

ah yeah sorry about that, let me add more detail :D

File: e2e_samples/temperature_events/LoadTesting/azure-pipeline.yml [hunk 1]
```diff
@@ -5,33 +5,39 @@ trigger:
...
-
+# Remember to modify the script paths. It starts from the root of your project.
 - task: AzureCLI@2
   displayName: 'Run IoTSimulator on ACI'
   inputs:
     azureSubscription: $(ServiceConnection)
     scriptType: pscore
     scriptPath: e2e_samples/temperature_events/LoadTesting/IoTSimulator.ps1
-    arguments: '-EventHubConnectionString "`"$(LoadTestEventHubConnectionString)`"" 
+    arguments: '-EventHubConnectionString "`"$(TemperatureDevice)`""
```
[the remark this answers is not in the document]
Comment (shervyna) [reply e2e_samples/temperature_events/LoadTesting/azure-pipeline.yml:40]: ah yes, thanks for catching :D

[the remark this answers is not in the document]
Comment (shervyna) [reply e2e_samples/temperature_events/LoadTesting/azure-pipeline.yml]: fixed :)

**Comment by shervyna** [comment]
@jmostella I've updated this PR with a brief readme. Can you help me review it again? thanks!

File: e2e_samples/temperature_events/LoadTesting/azure-pipeline.yml [hunk 2]
```diff
@@ -5,33 +5,39 @@ trigger:
...
-  MessageCount: 1000
-  Interval: 100
-  ContainerCount: 2
-  ResourceGroup: 'dataops-temperature-events'
-  EvhNamespace: 'evh-temperature-test'
-  EvhName: 'temperature-processing'
-  ServiceConnection: dataops-temperature-events-sn
-  KeyVaultName: 'kv-temperature-test'
+- group: load-testing-secrets
+- name: DeviceCount
```
Comment (akirakakar) [inline e2e_samples/temperature_events/LoadTesting/azure-pipeline.yml:17]: Should we just add these values in variable group as well?

**Review by akirakakar — APPROVED** [review]
Minor comment but lgtm.

File: e2e_samples/temperature_events/LoadTesting/azure-pipeline.yml (hunk 2 above)
Comment (DavidBurela) [reply e2e_samples/temperature_events/LoadTesting/azure-pipeline.yml:17]: Some are environment level that are dynamic in the Terraform.  Others are just specific knobs to tweak for the load test.  There are 7 variables left, of them:

- DeviceCount - Leave as pipeline variable
- MessageCount - Leave as pipeline variable
- Interval - Leave as pipeline variable
- ContainerCount - Leave as pipeline variable
- ResourceGroup - **Could move to variable group?** ✔
- ServiceConnection - Must be a pipeline variable ❌
- KeyVaultName - Must be a pipeline variable ❌

Comment (shervyna) [reply e2e_samples/temperature_events/LoadTesting/azure-pipeline.yml:17]: thanks for catching the keyvaultname var! let me remove it.

btw, does service connection has to be a pipeline variable(in yaml)? I think defining it in variable group works as well :D

File: e2e_samples/temperature_events/LoadTesting/README.md [hunk 3]
```diff
@@ -16,23 +16,22 @@ scriptPath: e2e_samples/temperature_events/LoadTesting/IoTSimulator.ps1
...
 
-Then, you can use the keyvault secrets directly in pipeline by specifying the variable group:
+- *Ingest-namespace*: eventhub namespace of the entry eventhub. 
+- *Ingest-name*: name of the entry eventhub. 
+- *Ingest-conn*: connection string of the entry eventhub.
+- *subscription-id*: subscription Id of the project.
+
+ ![var_group_secrets](../../../docs/images/var_group_secrets.png)
+
+Then, you can use these keyvault secrets directly in pipeline by specifying the variable group:
```
Comment (DavidBurela) [inline e2e_samples/temperature_events/LoadTesting/README.md:29]:
```suggestion
You are able to use these keyvault secrets directly in the pipeline, as the variable group is defined in the `azure-pipeline.yml` file
```

**Review by DavidBurela — APPROVED** [review]
Thanks for the updates that we discussed. Looks fine to me. I'm happy for you to complete the PR.  Any further enhancements can just be another PR.

**Review by jmostella — APPROVED** [review]

**Pull request merged**